applications/CompressiblePotentialFlowApplication/custom_processes/compute_embedded_lift_process.cpp

```diff
@@ -46,19 +46,15 @@ void ComputeEmbeddedLiftProcess<Dim, NumNodes>::Execute()
         auto it_elem=mrModelPart.ElementsBegin()+i;
         auto r_geometry = it_elem->GetGeometry();
 
-        BoundedVector<double,3> geometry_distances;
+        BoundedVector<double, NumNodes> elemental_distances;
         for(unsigned int i_node = 0; i_node<NumNodes; i_node++){
-            geometry_distances[i_node] = r_geometry[i_node].GetSolutionStepValue(GEOMETRY_DISTANCE);
+            elemental_distances[i_node] = r_geometry[i_node].GetSolutionStepValue(GEOMETRY_DISTANCE);
         }
-        bool is_embedded = PotentialFlowUtilities::CheckIfElementIsCutByDistance<Dim,NumNodes>(geometry_distances);
+        bool is_embedded = PotentialFlowUtilities::CheckIfElementIsCutByDistance<Dim,NumNodes>(elemental_distances);
 
         if (is_embedded && it_elem->Is(ACTIVE)){
-            array_1d<double,3> elemental_distances;
-            for(unsigned int i_node = 0; i_node<NumNodes; i_node++){
-                elemental_distances[i_node] = r_geometry[i_node].FastGetSolutionStepValue(GEOMETRY_DISTANCE);
-            }
 
-            const Vector& r_elemental_distances=elemental_distances;
+            const Vector& r_elemental_distances = elemental_distances;
```

Comment:
Why do you want this to be `const`?  It will be destroyed as soon as you leave the scope of the function.

Reply:
I use these distances to create the modified shape functions:

https://github.com/KratosMultiphysics/Kratos/blob/1fadba5aa272b5a5a86c7d59e432346533870f9f/applications/CompressiblePotentialFlowApplication/custom_processes/compute_embedded_lift_process.cpp#L57-L58

These distances are not meant to be modified that's why they are defined as `const` . Also I dont need them past this point. Do you think we can remove the const anyway?

Reply:
Considering that these elemental distances are an auxiliary variable I wouldn't create the `const` copy.